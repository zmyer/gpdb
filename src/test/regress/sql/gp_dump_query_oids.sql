-- Test the built-in gp_dump_query function.
SELECT gp_dump_query_oids('SELECT 123');
SELECT gp_dump_query_oids('SELECT * FROM pg_proc');
SELECT gp_dump_query_oids('SELECT length(proname) FROM pg_proc');

-- with EXPLAIN
SELECT gp_dump_query_oids('explain SELECT length(proname) FROM pg_proc');

-- with a multi-query statement
SELECT gp_dump_query_oids('SELECT length(proname) FROM pg_proc; SELECT abs(relpages) FROM pg_class');

-- Test error reporting on an invalid query.
SELECT gp_dump_query_oids('SELECT * FROM nonexistent_table');
SELECT gp_dump_query_oids('SELECT with syntax error');
