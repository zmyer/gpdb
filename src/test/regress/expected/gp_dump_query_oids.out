-- Test the built-in gp_dump_query function.
SELECT gp_dump_query_oids('SELECT 123');
      gp_dump_query_oids       
-------------------------------
 {"relids": "", "funcids": ""}
(1 row)

SELECT gp_dump_query_oids('SELECT * FROM pg_proc');
        gp_dump_query_oids         
-----------------------------------
 {"relids": "1255", "funcids": ""}
(1 row)

SELECT gp_dump_query_oids('SELECT length(proname) FROM pg_proc');
            gp_dump_query_oids             
-------------------------------------------
 {"relids": "1255", "funcids": "1317,406"}
(1 row)

-- with EXPLAIN
SELECT gp_dump_query_oids('explain SELECT length(proname) FROM pg_proc');
            gp_dump_query_oids             
-------------------------------------------
 {"relids": "1255", "funcids": "1317,406"}
(1 row)

-- with a multi-query statement
SELECT gp_dump_query_oids('SELECT length(proname) FROM pg_proc; SELECT abs(relpages) FROM pg_class');
                 gp_dump_query_oids                  
-----------------------------------------------------
 {"relids": "1255,1259", "funcids": "1317,406,1397"}
(1 row)

-- Test error reporting on an invalid query.
SELECT gp_dump_query_oids('SELECT * FROM nonexistent_table');
ERROR:  relation "nonexistent_table" does not exist
LINE 1: SELECT * FROM nonexistent_table
                      ^
QUERY:  SELECT * FROM nonexistent_table
SELECT gp_dump_query_oids('SELECT with syntax error');
ERROR:  syntax error at or near "with syntax"
LINE 1: SELECT with syntax error
               ^
QUERY:  SELECT with syntax error
